pkg/execution/state/redis_state/lua/consumePause.lua

```diff
@@ -4,66 +4,26 @@ Consumes a pause.
 
 Output:
   0: Successfully consumed
-  1: Pause not found
-
+  1: Pause already consumed
 ]]
 
--- The pause ID is always provided as a key, as is the lease ID.
-local pauseKey      = KEYS[1]
-local pauseStepKey  = KEYS[2]
-local pauseEventKey = KEYS[3]
-local pauseInvokeKey = KEYS[4]
-local actionKey     = KEYS[5]
-local stackKey      = KEYS[6]
-local keyMetadata   = KEYS[7]
-local keyPauseAddIdx = KEYS[8]
-local keyPauseExpIdx = KEYS[9]
-local keyRunPauses   = KEYS[10]
-
-local pauseID      = ARGV[1]
-local invokeCorrelationId = ARGV[2]
-local pauseDataKey = ARGV[3] -- used to set data in run state store
-local pauseDataVal = ARGV[4] -- data to set
+local actionKey     = KEYS[1]
+local stackKey      = KEYS[2]
+local keyMetadata   = KEYS[3]
 
-local pause = redis.call("GET", pauseKey)
-if pause == false or pause == nil then
-	-- Pause no longer exists.
-	if pauseEventKey ~= "" then
-		-- Clean up regardless
-		redis.call("HDEL", pauseEventKey, pauseID)
-		-- Add an index of when the pause was added.
-		redis.call("ZREM", keyPauseAddIdx, pauseID)
-		-- Add an index of when the pause expires.  This lets us manually
-		-- garbage collect expired pauses from the HSET below.
-		redis.call("ZREM", keyPauseExpIdx, pauseID)
-	end
-	return 1
-end
-
-redis.call("DEL", pauseKey)
```

Comment:
Since pauses are unsharded, we cannot delete pauses while updating function run state in the same script, as we can only access one Redis cluster per script. We have moved the pause cleanup logic into a separate step. While we lose atomicity, deleting a pause is idempotent, so this does not introduce any new issues.